frontend/src/queries/Query/Query.tsx

```diff
@@ -71,16 +72,92 @@ export function Query<Q extends Node>(props: QueryProps<Q>): JSX.Element | null
         if (propsQuery !== localQuery) {
             localSetQuery(propsQuery)
         }
-    }, [propsQuery])
+    }, [propsQuery]) // eslint-disable-line react-hooks/exhaustive-deps
 
     const query = readOnly ? propsQuery : localQuery
     const setQuery = propsSetQuery ?? localSetQuery
 
-    const queryContext = props.context || {}
+    const queryContext = useMemo(() => props.context || {}, [props.context])
```

Comment:
Do you know if this will gain you anything? It's just an object key lookup; I would guess that `useMemo` is doing more than that behind the scenes.

Reply:
This is because I wanna use `queryContext` as a dependency for the `useMemo` below. If we keep what we had before, then `{}` would always be a new object, therefore we wouldn't be caching anything

Reply:
Ah true, yes it would help a lot in that case!